### ISSUE TITLE
Missing sensor MK1? (question)

### ISSUE THREAD

**Issue by sshknoll** [BenPru/luxtronik#285]
Hi everyone,

first of thank you very much for the work you are putting into this project! :)

I recently installed the integration in HomeAssistant and it works quite well. What I am missing is the sensor values for MK1 mixing circuit (MK1 target and current). Are these deliberately not included or am I missing something here?

Thanks for any hints on that!

**Comment by toggm** [comment]
Same problem here. Tried to add them manually, but without any luck:
```
- platform: luxtronik2
  sensors:
   - group: calculations
     id: ID_WEB_Temperatur_TFB1
     friendly_name: Mischkreis1-Vorlauf
     icon: mdi:thermometer
   - group: calculations
     id: ID_WEB_Temperatur_TFB2
     friendly_name: Mischkreis2-Vorlauf
     icon: mdi:thermometer
   - group: calculations
     id: ID_WEB_Sollwert_TVL_MK1
     friendly_name: Mischkreis1 VL-Soll
     icon: mdi:thermometer
   - group: calculations
     id: ID_WEB_Sollwert_TVL_MK2
     friendly_name: Mischkreis2 VL-Soll
     icon: mdi:thermometer
```

**Comment by sshknoll** [comment]
As far as I understand, there is no use trying to add sensors anymore. Latest version doesn’t support Yaml config anymore.

There is a workaround by using the old plugin by Bouni in parallel.

But I was wondering if these sensors were intentionally left out.

**Comment by Mehli12** [comment]
Try to find this in the hided entities.

![Bildschirmfoto 2024-10-19 um 11 45 44](https://github.com/user-attachments/assets/656ace24-2abc-4f53-85e3-bd838a61928a)

**Comment by sshknoll** [comment]
Thanks, but I have already done that. There is no hidden entity that matches MK1 Temperature, only the curves for MK2 and MK3 can be accessed - not even their current temperatures. 
If an MK1 sensor existed, wouldn't it show up under hidden sensors, by the way?

<img width="347" alt="Bildschirmfoto 2024-10-19 um 17 41 53" src="https://github.com/user-attachments/assets/5b09a38b-dd65-4437-86c2-d61fdd140dab">

<img width="298" alt="Bildschirmfoto 2024-10-19 um 17 42 53" src="https://github.com/user-attachments/assets/fae5e43a-5b7c-4158-9734-f565cdd8eee2">

**Comment by Mehli12** [comment]
In my installation the MK1 = Heizkurve 1 = Heizkurve of the heat pump was directly discovered by the integration. MK2 is in my case for the curve for my floor heating. I don't use another curve because I have just 2 heating circuits. (heat pump it self and floor heating)

**Comment by toggm** [comment]
I only see those values regarding the measured temperatures:
![image](https://github.com/user-attachments/assets/2fdfbe00-368c-43ed-8500-d1fb68ca3c8b)

MK2 is only available to define the "Heizkurve" and is a static value over time:
![image](https://github.com/user-attachments/assets/1be187ae-aedd-4cab-a5d4-5cd44aeef713)

What I'm missing are those:
![image](https://github.com/user-attachments/assets/2610cfbe-bcdb-481a-93d2-fb0efe656879)

Defined here as:
![image](https://github.com/user-attachments/assets/a919086e-a77b-4636-abde-b3ac3cab2cb3)

**Comment by sshknoll** [comment]
Thank you for pointing that out with the screenshots. That's exactly what I am looking for! Any way to implement that. I'd be happy to help if somebody pointed me toward a way to do so. (which files?)

**Comment by toggm** [comment]
Wild guess would be to add a descriptor of those flags to https://github.com/BenPru/luxtronik/blob/main/custom_components/luxtronik/sensor_entities_predefined.py#L74

But didn't try it out yet...

**Comment by macrauder** [comment]
+1 Same behavior in my installation MK1 is missing

**Comment by Plobli** [comment]
For me, MK1 is called in Home Assistant heating curve 1 (Heizkurve 1) and is displayed. I only have an MK1 in the cooling area in the integration.

**Comment by toggm** [comment]
@Plobli But that's the "Heizkurve", which means the configuration of it and not the measured temperatur

**Comment by Firetribe** [comment]
I have the same issue... cannot add the mk1 or mk1_target, I don't have the entities. Would be nice to have to make sure I don't heat up my storage tank instead of the floors...

**Comment by rhammen** [comment]
@toggm @sshknoll @macrauder @Firetribe 
These sensors should now be present in pre-Release 2025.7.22-beta.
Can you confirm it now works ok, so we can close this issue?

**Comment by Firetribe** [comment]
@rhammen It’s there for me, perfect. Cannot yet verify the readings, since heater is off right now, but looks good 👍

**Comment by rhammen** [comment]
Closing as completed: fixed in 2025.7.22-beta.
